### PR TITLE
docs: Add library dependencies to README

### PR DESCRIPTION
**Pull request by coreh**
Hey there :wave: 

This PR documents the library dependencies that need to be installed in order for `cargo run` to succeed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,15 +29,29 @@ Also:
 
 Like others, Smithay as a compositor library has its own sample compositor: anvil.
 
-You can run it with cargo after having cloned this repository:
-
+You can run it with cargo after having cloned this repository and installed the required dependencies:
+
+- `libdbus-glib-1`
+- `libudev`
+- `libsystemd`
```
Comment (i509VCB) [inline README.md:36]: Not sure that systemd would apply to every udev session type?

Comment (coreh) [reply README.md:36]: It didn't compile without it for me, maybe it's picking it up from my environment somehow?

Are there other libraries that might be needed instead for non-systemd distros?

Comment (i509VCB) [reply README.md:36]: I don't know either, I'd probably need to spin up a bunch of vms to test this

Comment (vberger) [reply README.md:36]: The system dependencies would depend on which cargo features are enabled, `libsystemd` is pulled by `backend_session_logind`.

Comment (PolyMeilex) [reply README.md:36]: Same way [seatd/libseat](https://sr.ht/~kennylevinsen/seatd/), requires a libseat system lib as well

File: README.md [hunk 2]
```diff
@@ -29,15 +29,29 @@ Also:
...
+- `libxkbcommon`
+- `libinput`
+- `libgbm`
+
+```bash
```
Comment (PolyMeilex) [inline README.md:41]:
```suggestion
```bash
If you enable X11 support:
- `xwayland`
If you enable libseat (seatd, logind, elogind) support:
- [libseat](https://git.sr.ht/~kennylevinsen/seatd)
```

Comment (PolyMeilex) [reply README.md:41]: Wops, I meant for this to be above bash block, by bad

File: README.md [hunk 3]
```diff
@@ -29,15 +29,29 @@ Also:
...
+# (for Debian/Ubuntu)
```
Comment (PolyMeilex) [inline README.md:42]:
```suggestion
If you enable X11 support:
- `xwayland`
If you enable libseat (seatd, logind, elogind) support:
- [libseat](https://git.sr.ht/~kennylevinsen/seatd)

# (for Debian/Ubuntu)
```

File: README.md [hunk 4]
```diff
@@ -29,15 +29,29 @@ Also:
...
-
+You can run it with cargo after having cloned this repository and installed the required dependencies:
+
+- `libdbus-glib-1`
+- `libudev`
+- `libsystemd`
+- `libxkbcommon`
+- `libinput`
+- `libgbm`
+
```
Comment (PolyMeilex) [inline README.md:40]:
```suggestion

If you enable X11 support:
- `xwayland`
If you enable libseat (seatd, logind, elogind) support:
- [libseat](https://git.sr.ht/~kennylevinsen/seatd)
```

**Pull request closed without merging**